sdk/program/src/program.rs

```diff
@@ -54,13 +66,46 @@ pub fn invoke_signed(
     crate::program_stubs::sol_invoke_signed(instruction, account_infos, signers_seeds)
 }
 
-#[cfg(target_arch = "bpf")]
-extern "C" {
-    fn sol_invoke_signed_rust(
-        instruction_addr: *const u8,
-        account_infos_addr: *const u8,
-        account_infos_len: u64,
-        signers_seeds_addr: *const u8,
-        signers_seeds_len: u64,
-    ) -> u64;
+/// Maximum size that can be set using sol_set_return_data()
+pub const MAX_RETURN_DATA: usize = 1024;
+
+/// Set a program's return data
+pub fn set_return_data(data: &[u8]) {
+    #[cfg(target_arch = "bpf")]
+    {
+        extern "C" {
+            fn sol_set_return_data(data: *const u8, length: u64);
+        }
+
+        unsafe { sol_set_return_data(data.as_ptr(), data.len() as u64) };
+    }
+
+    #[cfg(not(target_arch = "bpf"))]
+    crate::program_stubs::sol_set_return_data(data)
+}
+
+/// Get the return data from invoked program
+pub fn get_return_data() -> Option<(Pubkey, Vec<u8>)> {
+    #[cfg(target_arch = "bpf")]
+    {
+        extern "C" {
+            fn sol_get_return_data(data: *mut u8, length: u64, program_id: *mut Pubkey) -> u64;
+        }
+
+        let mut buf = [0u8; MAX_RETURN_DATA];
+        let mut program_id = Pubkey::default();
+
+        let size =
+            unsafe { sol_get_return_data(buf.as_mut_ptr(), buf.len() as u64, &mut program_id) };
+
+        if size == 0 {
+            None
+        } else {
+            let size = std::cmp::min(size as usize, MAX_RETURN_DATA);
```

Comment:
nit: use a `use` for `std::cmp`

Reply:
I agree having the full namespace there is a bit much. Since this is in a target == bpf block, we can't have a `use std::cmp::min;` at the top without `#[allow(unused)]'. So I've put the use in the same block.